Add support for focusing runtime entities, fix focus for baked entities

germanoeich opened the pull request:
### Base of Changes

0.12.0-beta8

### Review Items

- [x] Regression concerns

- [x] Functional correctness (code-read only)

- [x] Use-case flexibility

- [x] Performance considerations

- [x] Naming, comprehension, and code aesthetic (aside from white-space)

### Preferred Contributor Name and Shoutouts

github name is fine

## Details

When using QVVS or unity transforms and generating entities at runtime, with no baked gameobject counter-part, focusing on the selected entity in the hierarchy with the F shortcut didn't work. This is particularly annoying for procedurally generated entities. With a baked entity, focusing would take you to the GameObject position (origin of the entity), instead of the current transform position. This caused focus lock (shift + f) to not work and focus to be rather useless if your entities change position.

This PR fixes all those cases. When focusing a baked entity, the scene camera will focus on the current entity position according to its transform. Runtime only entities are also handled and focus and focus lock work on them now.

The packages added to the asmdef are: `Unity.Mathematics.Extensions` (For AABB), `Unity.Entities.Graphics` (For WorldRenderBounds), `Latios.Transforms`, `Unity.Entities.Editor` and `Latios.Core` (For LatiosWorld).

Dreaming381 commented on Transforms/CachedQvvs/Editor/QvvsSceneViewFocus.cs:
```diff
@@ -0,0 +1,41 @@
+using UnityEditor;
+using UnityEngine;
+
+static class QvvsSceneViewFocus
```
This needs to also work in Unity Transforms mode. And should probably be general purpose for that.

I think this editor script should be moved to Kinemation's Editor and this type should be renamed to `RuntimeEntitySceneViewFocus`.

Dreaming381 commented on Transforms/CachedQvvs/Editor/QvvsSceneViewFocus.cs:
```diff
@@ -0,0 +1,41 @@
+            var bounds = w.Unmanaged.EntityManager.GetComponentData<WorldRenderBounds>(entity).Value;
+            sceneView.Frame(new Bounds(bounds.Center, bounds.Size), false);
+        }
+        else if (em.HasComponent<TransformAspect>(entity))
```
This should probably be 
```csharp
em.HasComponent(entity, Latios.Transforms.Abstract.QueryExtensions.GetAbstractWorldTransformROComponentType())
```

Dreaming381 commented on Transforms/CachedQvvs/Editor/QvvsSceneViewFocus.cs:
```diff
@@ -0,0 +1,41 @@
+            Selection.activeGameObject != null) return;
+            
+        var entity = selectionProxy.Entity;
+        var w      = selectionProxy.World;
+        var em     = w.EntityManager;
```
Nit: Just `var em = selectionProxy.World.EntityManager` is sufficient.

Dreaming381 replied:
Actually, now that I think about it, we should probably check that the `World` is a `LatiosWorld`, and if not, `return` here to preserve "inert behavior until a bootstrap is installed".

Dreaming381 commented on Transforms/CachedQvvs/Editor/QvvsSceneViewFocus.cs:
```diff
@@ -0,0 +1,41 @@
+
+        if (em.HasComponent<WorldRenderBounds>(entity))
+        {
+            var bounds = w.Unmanaged.EntityManager.GetComponentData<WorldRenderBounds>(entity).Value;
```
Nit: Replace `w.Unmanaged.EntityManager` with `em`.

Dreaming381 commented on Transforms/CachedQvvs/Editor/QvvsSceneViewFocus.cs:
```diff
@@ -0,0 +1,41 @@
+        }
+        else if (em.HasComponent<TransformAspect>(entity))
+        {
+            var t = em.GetAspect<TransformAspect>(entity);
+            sceneView.LookAt(t.worldPosition, sceneView.rotation);
```
You want `WorldTransformReadOnlyAspect` here.

Dreaming381 reviewed (commented):
This seems like a good quality of life improvement. Thank you for submitting!

germanoeich commented:
@Dreaming381 Thank you for the review! I believe I've fixed the points you outlined, let me know if further changes are needed!

I took a little time to look into locking entities and implemented that too. The PR description is updated with the changes
